test/e2e/scheduling/predicates.go

```diff
@@ -1056,7 +1056,6 @@ func GetPodsScheduled(workerNodes sets.String, pods *v1.PodList) (scheduledPods,
 			_, scheduledCondition := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
 			framework.ExpectEqual(scheduledCondition != nil, true)
 			if scheduledCondition != nil {
-				framework.ExpectEqual(scheduledCondition.Status, v1.ConditionFalse)
```

Comment:
@Huang-Wei any idea why we had this check in the first place? if a pod was never tried, scheduled condition would never be set, which should be a valid notScheduledPod case.

Reply:
@ahg-g I also was wondering about the following line, it will not add the pod to the list of unscheduled unless its reason is `Unschedulable`. It seems that a pod should also be added to that list if it is schedulable but just no scheduled yet for some reason. I may be missing context though

Reply:
@ahg-g I've no idea either. It looks like that check is incorrect, but seems we don't have points reporting errors on that line.

@hasheddan Are you sure the flake turns out that the condition is Unknown instead of Scheduled? I asked this is b/c I feel like a scheduled Pod on master node can fall into current `else` block. If so, we should rule master pods out by changing `else` to `else if pod.Spec.NodeName == ""`.

Reply:
@Huang-Wei I can update, but I am not sure if it makes a difference because the next line checks `if scheduledCondition.Reason == "Unschedulable"`. so `Scheduled` pods would be ignored anyway. Is that how you see it as well?

Reply:
If we made it `else if pod.Spec.NodeName == ""`, then it cannot be a `Scheduled` pod. But it's still possible (1) the status is Unknown, (2) the status is nil. So, a bit aggressive, how about changing the whole function to:

```go
// GetPodsScheduled returns a number of currently scheduled and not scheduled Pods on worker nodes.
func GetPodsScheduled(workerNodes sets.String, pods *v1.PodList) (scheduledPods, notScheduledPods []v1.Pod) {
	for _, pod := range pods.Items {
		if pod.Spec.NodeName != "" && workerNodes.Has(pod.Spec.NodeName) {
			_, scheduledCondition := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
			framework.ExpectEqual(scheduledCondition != nil, true)
			if scheduledCondition != nil {
				framework.ExpectEqual(scheduledCondition.Status, v1.ConditionTrue)
				scheduledPods = append(scheduledPods, pod)
			}
		} else if pod.Spec.NodeName == "" {
			notScheduledPods = append(notScheduledPods, pod)
		}
	}
	return
}
```

Reply:
@Huang-Wei this LGTM, updating now :)

Reply:
Thanks. @ahg-g wanna take another look?